keras_core/models/functional.py

```diff
@@ -459,9 +464,14 @@ def process_layer(layer_data):
             layer_name = layer_data["name"]
 
             # Instantiate layer.
-            layer = serialization_lib.deserialize_keras_object(
-                layer_data, custom_objects=custom_objects
-            )
+            if "module" not in layer_data:
+                layer = saving_utils.model_from_config(
```

Comment:
Add comment that this is for the legacy case

Reply:
Comment added, fixed!